test_runner/fixtures/pg_version.py

```diff
@@ -0,0 +1,57 @@
+import enum
+import os
+from typing import Iterator
+
+import pytest
+from _pytest.config.argparsing import Parser
+from pytest import FixtureRequest
+
+from fixtures.log_helper import log
+
+"""
+This fixture is used to determine which version of Postgres to use for tests.
+"""
+
+
+# Inherit PgVersion from str rather than int to make it easier to pass as a command-line argument
+# TODO: use enum.StrEnum for Python >= 3.11
+@enum.unique
+class PgVersion(str, enum.Enum):
+    V14 = "14"
+    V15 = "15"
+    # Instead of making version an optional parameter in methods, we can use this fake entry
+    # to explicitly rely on the default server version (could be different from pg_version fixture value)
+    NOT_SET = "<-POSTRGRES VERSION IS NOT SET->"
+
+    @staticmethod
+    def from_int(int_version: int) -> "PgVersion":
```

Comment:
what do you think about using

```python
from __future__ import annotations
```
?

I'm not insisting, just asking

Reply:
Eventually, we should update Python to 3.11 and then we can use `Self` annotation.
So I'd keep it as is for now.